agent/agentsocket/socket_windows.go

```diff
@@ -0,0 +1,53 @@
+//go:build windows
+
+package agentsocket
+
+import (
+	"net"
+	"os"
+	"time"
+
+	"golang.org/x/xerrors"
+)
+
+// createSocket returns an error indicating that agentsocket is not supported on Windows.
+// This feature is unix-only in its current experimental state.
+func createSocket(path string) (net.Listener, error) {
+	return nil, xerrors.New("agentsocket is not supported on Windows")
```

Comment:
IIRC supporting Windows should be possible, at least to some degree. We have other socket support in the product for Windows but can't recall if it has limitations.